src/mono/wasm/runtime/gc-handles.ts

```diff
@@ -131,3 +131,9 @@ export function _lookup_js_owned_object(gc_handle: GCHandle): any {
     }
     return null;
 }
+
+/// Called from the C# threadpool worker loop to find out if there are any
+/// unsettled JS promises that need to keep the worker alive
+export function mono_wasm_eventloop_has_unsettled_interop_promises(): boolean {
+    return _js_owned_object_table.size > 0;
```

Comment:
This table also has all other types of objects in it, not just promises. We may need to track promises separately. The `JSHandle` still needs to be thread-unique, not just unique per type.

Reply:
> > This table also has all other types of objects in it, not just promises.
> 
> I see. So it is probably better to just augment `_marshal_task_to_cs` to increment/decrement an unsettled promises counter.

yes, when in MT build.

> > The `JSHandle` still needs to be thread-unique, not just unique per type.
> 
> Sorry, I don't understand what this means.

The other implementation option could be to have table per type. But allocation of the JSHandle would have to be common.